RadixWallet/Core/FeaturePrelude/AddressView/AddressDetails.swift

```diff
@@ -99,51 +106,42 @@ public struct AddressDetails: Sendable, FeatureReducer {
 		}
 	}
 
-	private func loadTitle(state: inout State) -> Effect<Action> {
+	private func loadTitleEffect(state: inout State) -> Effect<Action> {
 		state.title = .loading
 		return .run { [address = state.address] send in
-			let result: TaskResult<String?> = switch address {
-			case let .account(address, _):
-				await TaskResult {
-					let account = try await accountsClient.getAccountByAddress(address)
-					return account.displayName.rawValue
-				}
-			case let .resource(address):
-				await TaskResult {
-					let resource = try await onLedgerEntitiesClient.getResource(address)
-					return resource.resourceTitle
-				}
-			case let .validator(address):
-				await TaskResult {
-					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
-					return entity.metadata?.name
-				}
-			case let .package(address):
-				await TaskResult {
-					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
-					return entity.metadata?.name
-				}
-			case let .resourcePool(address):
-				await TaskResult {
-					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
-					return entity.metadata?.name
-				}
-			case let .component(address):
-				await TaskResult {
-					let entity = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .resourceMetadataKeys)
-					return entity.metadata?.name
-				}
-			case let .nonFungibleGlobalID(globalId):
-				await TaskResult {
-					let resource = try await onLedgerEntitiesClient.getResource(globalId.resourceAddress)
-					return resource.resourceTitle
-				}
-			}
-			await send(.internal(.loadedTitle(result)))
+			await send(.internal(.loadedTitle(TaskResult {
+				try await loadTitle(address: address)
+			})))
+		}
+	}
+
+	private func loadTitle(address: LedgerIdentifiable.Address) async throws -> String? {
+		switch address {
+		case let .account(address, _):
+			let account = try await accountsClient.getAccountByAddress(address)
```

Comment:
I suppose it's a matter of taste but it looks like all these branches could be written as one-liners, without a `return`.
```
try await accountsClient.getAccountByAddress(address).displayName.rawValue
```